src/itwinai/torch/distributed.py

```diff
@@ -1010,3 +1012,177 @@ def gather(self, tensor: torch.Tensor, dst_rank: int = 0):
             list[Any]: input object wrapped in a list.
         """
         return [tensor]
+
+
+class RayDistributedStrategy(TorchDistributedStrategy):
+    def _initialize_ray(self) -> None:
+        try:
+            ip_head = os.environ.get("ip_head")
+            head_node_ip = os.environ.get("head_node_ip")
+
+            if not ip_head or not head_node_ip:
+                raise EnvironmentError(
+                    "Ray initialization requires 'ip_head' and 'head_node_ip' to be set."
+                )
+
+            if not ray.is_initialized():
+                ray.init(address="auto")
+                print("Ray initialized!")
```

Comment:
Not a big deal, but you could do the check `is_initialized()` at the beginning and return from the function if it isn't.

Reply:
Added!